src/main/java/bio/terra/cli/apps/LocalProcessCommandRunner.java

```diff
@@ -0,0 +1,53 @@
+package bio.terra.cli.apps;
+
+import bio.terra.cli.apps.utils.LocalProcessLauncher;
+import bio.terra.cli.context.GlobalContext;
+import bio.terra.cli.context.WorkspaceContext;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** This class runs client-side tools in a local process. */
+public class LocalProcessCommandRunner extends CommandRunner {
+  private static final Logger logger = LoggerFactory.getLogger(LocalProcessCommandRunner.class);
+
+  private LocalProcessLauncher localProcessLauncher = new LocalProcessLauncher();
+
+  public LocalProcessCommandRunner(GlobalContext globalContext, WorkspaceContext workspaceContext) {
+    super(globalContext, workspaceContext);
+  }
+
+  /**
+   * Run a tool command in a local process.
+   *
+   * <p>Some setup/cleanup commands for gcloud authentication and configuration will be run
+   * before/after the given command.
+   *
+   * @param command the command and arguments to execute
+   * @param envVars a mapping of environment variable names to values
+   */
+  protected void runToolCommandImpl(List<String> command, Map<String, String> envVars) {
+    List<String> bashCommands = new ArrayList<>();
+    bashCommands.add("gcloud config set project " + workspaceContext.getGoogleProject());
+    bashCommands.add(
+        "gcloud auth activate-service-account --key-file=$GOOGLE_APPLICATION_CREDENTIALS");
+    bashCommands.add(buildFullCommand(command));
+
+    List<String> processCommand = new ArrayList<>();
+    processCommand.add("bash");
+    processCommand.add("-c");
+    processCommand.add(String.join("; ", bashCommands));
+
+    // launch the child process
+    localProcessLauncher.launchProcess(processCommand, envVars);
+
+    // stream the output to stdout/err
+    localProcessLauncher.streamOutputForProcess();
+
+    // block until the child process exits
+    int exitCode = localProcessLauncher.waitForTerminate();
```

Comment:
Is there a reason to split out 3 public methods for localProcessLauncher, or should they always be called together?

Reply:
At the moment, they should always be called together. I separated them out because:
- I think it's easier to follow the main logic of what's happening, rather than concatenating it all in one method.
- In the future, we may want to keep a single local process running and periodically send new commands to it. This would be the analog of doing the same with a single container. It would eliminate the start-up costs that come with creating a new process or container each time. This is something we've discussed with the solutions team before (`terra shell` and `terra gsutil/nextflow/...` performance conversations). If we take that route, then I think these methods may be called separately.

Reply:
nit: I'd bias towards keeping the public api simple for now and changing it when/if we have to under general YAGNI principles. I think the three methods are helpful for breaking up what's happening, so let's call all 3 as private methods from a single public method.